diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +122,168 @@ class LineLength : Rule("line-length") {
         }
     }
 
+    private fun fixError(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        if (wrongNode.elementType == EOL_COMMENT) {
+            fixComment(wrongNode, configuration)
+            return
+        }
+        var parent = wrongNode
+        do {
+            when (parent.elementType) {
+                FUN -> {
+                    if (!parent.hasChildOfType(BLOCK))
+                        parent.addChild(PsiWhiteSpaceImpl("\n"), parent.findChildByType(EQ)!!.treeNext)
+                    return
+                }
+                CONDITION -> {
+                    fixCondition(parent, configuration)
+                    return
+                }
+                PROPERTY -> {
+                    fixProperty(parent, configuration)
+                    return
+                }
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(wrongNode.startOffset).second
+        if (leftOffset >= configuration.lineLength)
```

Comment:
Could this actually happen if you retrieve `wrongNode` as node at offset equal to maximum length? Could you provide example of such case?

Reply:
No, I can't. My mistake